### DISCUSSION TITLE
community visibility (roadmap): where the core team of prest should invest their efforts in 2023

### DISCUSSION THREAD

**Discussion by avelino** [prest/prest#759] [General]
Thread to discuss where we should put **efforts in 2023**

I created a [milestone](https://github.com/prest/prest/milestone/20) to add the issues that are (or will be) marked as planned for 2023

> the 1 year release cycle is too big, I wish in the future we had a 3 - 4 month release cycle (planning), but starting with 1 year is the best we can do today - I believe it is possible, I count on your dedication to understand :D

if you use this thread to **discuss our future** _(I am talking about 2023)_, I put what you believe we should invest efforts in **this year**

**Comment by vmesel** [comment]
We need to solve our release issue on Docker and GHCR. This is a must for this year.

The other topics I would like us to address are:

 - Testing environment fixed and easier to use
 - Multi-database support
 - Redshift adapter

**Reply by arxdsilva** [reply]
> - Testing environment fixed and easier to use

Agree, we should spend some time there, we need this to be as smooth as possible

> - Multi-database support

Not sure how we can do this without breaking things, now it is possible but not advised. I believe that more things may have priority.

> - Redshift adapter

This will be some work but maybe worth if we get more of the community supporting this.

Also would be good (not in this order necessarely):
1. support for pagination;
2. mutation logs table;
3. have a separate prest management DB from the managed API DB's;
4. better envs / configuration;
5. extended set of plugin interface;
6. separate auth into the new DB and allow permissioning;
7. observability upgrades - prometheus integration for eg.

**Reply by vmesel** [reply]
Fix our versioning CLI output. We are always in the same version, even if we update the tag.
